### ISSUE TITLE
Prod frontend build failed with Nextjs Image component

### ISSUE THREAD

**Issue by sky4git** [aws-amplify/amplify-js#7547]
**Describe the bug**
NextJS recent version 10.0.0 included the **Image** component which does the image optimization automatically. If we use it our NextJS app then frontend build at the deployment with the following error:
Error: Image Optimization using Next.js' default loader is not compatible with `next export`.
Possible solutions:
  - Use `next start`, which starts the Image Optimization API.
  - Use Vercel to deploy, which supports Image Optimization.
  - Configure a third-party loader in `next.config.js`.

**To Reproduce**
1. Create a sample NextJS app with latest version.
2. Include Image component.
3. Do the frontend build as per the https://docs.amplify.aws/guides/hosting/nextjs/q/platform/js#git-based-deployments

**Expected behavior**
No error should happen or Amplify need to update documentation on what to do when we use Image component.

**Code Snippet**
Please provide a code snippet or a link to sample code of the issue you are experiencing to help us reproduce the issue. (Be sure to remove any sensitive data)
![image](https://user-images.githubusercontent.com/8438024/104256356-30e1cd00-54cf-11eb-80d7-c7e85d4ab963.png)

**Screenshots**
![image](https://user-images.githubusercontent.com/8438024/104256150-bf098380-54ce-11eb-843e-f3b51a395811.png)

**Comment by bobbyhadz** [comment]
The next.js `Image` component introduces some optimizations at **runtime**, which means a server is required. 

Amplify only supports hosting static sites with `next export`, therefore you can't use the `Image` component.

**Comment by sammartinez** [comment]
Thank you for this feedback @sky4git, @bobbyhadz is correct that we currently only support SSR on the client. I have marked this as a feature request.

**Comment by KlodAyzak** [comment]
@bobbyhadz, do you have any example to share? I have the same issue and I am not able to solve it it.

**Comment by bobbyhadz** [comment]
@KlodAyzak the next.js `Image` component is currently not support when deploying with amplify

**Comment by cmnstmntmn** [comment]
any workarounds around this issue?

**Comment by cmnstmntmn** [comment]
https://aws.amazon.com/about-aws/whats-new/2021/08/aws-amplify-hosting-support-next-js-version-11/

Image optimization is not working

**Comment by eveevans** [comment]
I dont understand... the documentation says this: `Amplify Hosting supports all the latest Next.js features including incremental static regeneration, automatic image optimization, and script optimization` but still gives me the same error

**Comment by eveevans** [comment]
@cmnstmntmn  did you solve this ?

**Comment by prestonholt** [comment]
My Amplify build (SSR) succeeds with the Image component, but the images fail to load on the live site. It's working in my dev environment

**Comment by eveevans** [comment]
In my case, was a permission issue on the AWS role (the default Amplify role didnt work, I had to give it more permissions) 
Now my  NextJS 11, on Amplify with a SSG site its working fine. 
But the Image behavior its kind of buggy.

**Comment by cmnstmntmn** [comment]
@eveevans did you manage to get next-gen image optimization?

**Comment by cmnstmntmn** [comment]
i'll link this issue i posted on Next JS repo https://github.com/vercel/next.js/issues/28241

**Comment by eveevans** [comment]
@cmnstmntmn Yes, basically I'm using NextJS 11.1 and all the default configs, except for the Next.js version on amplify in which I changed `9` for `latest` 

Hope you this help you: 

Framework
```
Framework
Next.js - SSR
```

Next.js version `latest`

script config: 
```
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
  },
```

Dependencies: 
```
  "dependencies": {
    "next": "11.1.0",
    "react": "17.0.2",
    "react-dom": "17.0.2",
  },
```

**Comment by cmnstmntmn** [comment]
@eveevans indeed, but for reason or another the image is not converted to next-gen format.

1. this is the Vercel, all ok https://nextjs-test-seven-ashy.vercel.app/ (the image is converted to webp)
2. same code on AWS: https://main.d79bb4yng9r5p.amplifyapp.com/ (the image is not converted to webp)

**Comment by carlosriveroib** [comment]
> @eveevans indeed, but for a reason or another the image is not converted to the next-gen format.
> 
> 1. this is the Vercel, all ok https://nextjs-test-seven-ashy.vercel.app/ (the image is converted to webp)
> 2. same code on AWS: https://main.d79bb4yng9r5p.amplifyapp.com/ (the image is **NOT** converted to webp)

Yes, for me is not converting images too.

On the other hand, I'm using NextJS i18n https://nextjs.org/docs/advanced-features/i18n-routing
I have the default locale configured in `en` and my browser is configured in `es`. It should redirect to `es` automatically. It works on Vercel but not in AWS
